architecture/2024-08-dashboard-component-threat-model.md

```diff
@@ -0,0 +1,299 @@
+# Radius Dashboard Component Threat Model
+
+- **Author**: @nithyatsu
+
+## Overview
+
+This document provides a threat model for the Radius Dashboard component. It identifies potential security threats to this part of Radius and suggests possible mitigations. The document includes an analysis of the system, its assets, identified threats, and recommended security measures to protect the system.
+
+The Radius Dashboard component provides the frontend experience for Radius. 
+It provides visual and textual representation of user's applications, environments and recipes.
+
+## Terms and Definitions
+
+| Term                  | Definition      |
+| --------------------- | ----------------------------- |
+| mTLS | Mutual Transport Layer Security (mTLS) allows two parties to authenticate each other during the initial connection of an SSL/TLS handshake. |
+| UCPD | Universal Control Plane Daemon for Radius |
+| DoS | Denial of Service |
+| SPA | Single Page Application (SPA): Typically a React application that contains the logic for most user interactions within the React components. SPAs do not require reloading the page from the server during user interactions. |
+
+## System Description
+
+The Dashboard component is an SPA built as [Backstage](https://backstage.io/) plugin. It is a client of Radius API. It queries the graph of an application or a list of environment and constructs a visual representation of the response.
+
+### Architecture
```

Comment:
Are we thinking that the dashboard is an extension point for customers? If so, does that affect the threat model in any way?

Reply:
Dashboard does not have extendability.